### PR TITLE
PMM-3716 PMM2-API Tests Job

### PR DESCRIPTION
**Pull request by puneet0191**

### CODE REVIEW

File: pmm/pmm2-api-tests.groovy [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+                    sudo mv go /usr/local
+                    export GOROOT=/usr/local/go
+                    export GOPATH=/usr/local/go
+                    export PATH=$GOPATH/bin:$GOROOT/bin:$PATH
+                    go env
+                    sudo ln -s \$(pwd -P) $GOPATH/src/pmm-api-tests
+                    sudo curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
+                    cd $GOPATH/src/pmm-api-tests/
+                    dep ensure -v
+                    make
```
Comment (AlekSi) [inline pmm/pmm2-api-tests.groovy]: @puneet0191 @BupycHuk Let's dockerise our tests instead so they can be run with just `make run` or something without installing Go, etc.

Comment (AlekSi) [reply pmm/pmm2-api-tests.groovy]: https://jira.percona.com/browse/PMM-3719

File: pmm/pmm2-api-tests.groovy [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+            }
+        }
+        stage('Run API Test') {
+            steps {
+                withCredentials([sshUserPrivateKey(credentialsId: 'aws-jenkins', keyFileVariable: 'KEY_PATH', passphraseVariable: '', usernameVariable: 'USER')]) {
+                    sh '''
+                        sudo yum -y install docker
+                        sudo usermod -aG docker ec2-user
+                        sudo service docker start
+                        sudo docker build -t pmm-api-tests .
```
Comment (hors) [inline pmm/pmm2-api-tests.groovy]: Why do you build docker image every time when you run the job? Maybe it is better to build it once and push it to docker hub (Percona Lab) .

Comment (AlekSi) [reply pmm/pmm2-api-tests.groovy]: IMO, that okay: we only put Go binaries there, that does not take much time, and they (test binaries) change often.

File: pmm/pmm2-api-tests.groovy [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+            }
+        }
+        stage('Start staging') {
+            steps {
+                runStaging(DOCKER_VERSION, CLIENT_VERSION, '--addclient=ps,1')
+            }
+        }
+        stage('Sanity check') {
+            steps {
+                sh "curl --silent --insecure '${PMM_URL}/prometheus/targets' | grep localhost:9090"
```
Comment (AlekSi) [inline pmm/pmm2-api-tests.groovy]: It is better to use `/ping` API – it returns 200 if PostgreSQL, Prometheus, etc. are up.

File: pmm/pmm2-api-tests.groovy [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+        disableConcurrentBuilds()
+    }
+    triggers {
+        upstream upstreamProjects: '', threshold: hudson.model.Result.SUCCESS
+    }
+
+    stages {
+        stage('Prepare') {
+            steps {
+                // clean up workspace and fetch pmm-qa repository
```
Comment (AlekSi) [inline pmm/pmm2-api-tests.groovy]: But that's not pmm-qa repo

File: pmm/pmm2-api-tests.groovy [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+            }
+        }
+        stage('Sanity check') {
+            steps {
+                sh "curl --silent --insecure '${PMM_URL}/prometheus/targets' | grep localhost:9090"
+            }
+        }
+        stage('Sleep') {
+            steps {
+                sleep 100
```
Comment (AlekSi) [inline pmm/pmm2-api-tests.groovy]: Once `/ping` returns 200 (that could take a while, we should try it in loop), there is no longer a need for sleep.

Comment (AlekSi) [reply pmm/pmm2-api-tests.groovy]: Alternatively, we may install Docker, etc., and then check /ping – to do more things in parallel, and reduce total time.

File: pmm/pmm2-api-tests.groovy (hunk 4 above)
Comment (puneet0191) [reply pmm/pmm2-api-tests.groovy]: it will be in the future, once we merge changes to api-tests

File: pmm/pmm2-api-tests.groovy (hunk 5 above)
Comment (puneet0191) [reply pmm/pmm2-api-tests.groovy]: hmm, sure makes sense. I checked `ping` this morning, doesn't seem to work, comes back with an empty response, `http://server_url/ping` right?, comes back as `{}`

Comment (AlekSi) [reply pmm/pmm2-api-tests.groovy]: That's expected. What is important is status code – 200 if everything is ready.

File: pmm/pmm2-api-tests.groovy (hunk 4 above)
Comment (AlekSi) [reply pmm/pmm2-api-tests.groovy]: But then we can update comment too. I don't think that is a good idea to have a comment in sync with _future_ code, not with the current one. :)

**Pull request merged**